diff -ru old/Configure new/Configure
--- old/Configure	2008-09-18 14:56:10.000000000 +0200
+++ new/Configure	2009-03-10 23:48:48.034829900 +0100
@@ -531,7 +531,7 @@
 "BC-32","bcc32::::WIN32::BN_LLONG DES_PTR RC4_INDEX EXPORT_VAR_AS_FN:${no_asm}:win32",
 
 # MinGW
-"mingw", "gcc:-mno-cygwin -DL_ENDIAN -DWIN32_LEAN_AND_MEAN -fomit-frame-pointer -O3 -march=i486 -Wall::-D_MT:MINGW32:-lws2_32 -lgdi32 -lcrypt32:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts} EXPORT_VAR_AS_FN:${x86_asm}:coff:win32:cygwin-shared:-D_WINDLL -DOPENSSL_USE_APPLINK:-mno-cygwin:.dll.a",
+"mingw", "gcc:-g -DOPENSSL_SYSNAME_WIN32 -DL_ENDIAN -DWIN32_LEAN_AND_MEAN -fomit-frame-pointer -O3 -march=i486 -Wall::-D_MT:MINGW32:-lws2_32 -lgdi32 -lcrypt32:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts} EXPORT_VAR_AS_FN:${x86_asm}:coff:win32:cygwin-shared:-D_WINDLL -DOPENSSL_USE_APPLINK::.dll.a",
 # As for OPENSSL_USE_APPLINK. Applink makes it possible to use .dll
 # compiled with one compiler with application compiled with another
 # compiler. It's possible to engage Applink support in mingw64 build,
@@ -539,7 +539,7 @@
 # handling, one can't seriously consider its binaries for using with
 # non-mingw64 run-time environment. And as mingw64 is always consistent
 # with itself, Applink is never engaged and can as well be omitted.
-"mingw64", "gcc:-mno-cygwin -DL_ENDIAN -O3 -Wall -DWIN32_LEAN_AND_MEAN -DUNICODE -D_UNICODE::-D_MT:MINGW64:-lws2_32 -lgdi32 -lcrypt32:SIXTY_FOUR_BIT RC4_CHUNK_LL DES_INT EXPORT_VAR_AS_FN:${x86_64_asm}:mingw64:win32:cygwin-shared:-D_WINDLL:-mno-cygwin:.dll.a",
+"mingw64", "gcc:-g -DOPENSSL_SYSNAME_WIN32 -DL_ENDIAN -O3 -Wall -DWIN32_LEAN_AND_MEAN -DUNICODE -D_UNICODE::-D_MT:MINGW64:-lws2_32 -lgdi32 -lcrypt32:SIXTY_FOUR_BIT RC4_CHUNK_LL DES_INT EXPORT_VAR_AS_FN:${x86_64_asm}:mingw64:win32:cygwin-shared:-D_WINDLL::.dll.a",
 
 # UWIN 
 "UWIN", "cc:-DTERMIOS -DL_ENDIAN -O -Wall:::UWIN::BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:${no_asm}:win32",
diff -ru old/crypto/dso/dso_win32.c new/crypto/dso/dso_win32.c
--- old/crypto/dso/dso_win32.c	2008-09-18 14:56:10.000000000 +0200
+++ new/crypto/dso/dso_win32.c	2009-03-10 23:48:48.034829900 +0100
@@ -60,6 +60,7 @@
 #include <string.h>
 #include "cryptlib.h"
 #include <openssl/dso.h>
+#include <windows.h>
 
 #if !defined(DSO_WIN32)
 DSO_METHOD *DSO_METHOD_win32(void)
diff -ru old/ms/do_nasm.bat new/ms/do_nasm.bat
--- old/ms/do_nasm.bat	2008-09-18 14:56:10.000000000 +0200
+++ new/ms/do_nasm.bat	2009-03-10 23:48:48.034829900 +0100
@@ -1,6 +1,7 @@
 
 perl util\mkfiles.pl >MINFO
 perl util\mk1mf.pl nasm VC-WIN32 >ms\nt.mak
+perl util\mk1mf.pl debug nasm VC-WIN32 >ms\ntdebug.mak
 perl util\mk1mf.pl dll nasm VC-WIN32 >ms\ntdll.mak
 perl util\mk1mf.pl nasm BC-NT >ms\bcb.mak
 
diff -ru old/ms/do_win64a.bat new/ms/do_win64a.bat
--- old/ms/do_win64a.bat	2005-07-05 02:24:12.000000000 +0200
+++ new/ms/do_win64a.bat	2009-03-10 23:57:55.316079900 +0100
@@ -13,6 +13,7 @@
 
 :proceed
 perl util\mk1mf.pl VC-WIN64A >ms\nt.mak
+perl util\mk1mf.pl debug VC-WIN64A >ms\ntdebug.mak
 perl util\mk1mf.pl dll VC-WIN64A >ms\ntdll.mak
 
 perl util\mkdef.pl 32 libeay > ms\libeay32.def
diff -ru old/util/pl/VC-32.pl new/util/pl/VC-32.pl
--- old/util/pl/VC-32.pl	2008-11-18 23:23:20.000000000 +0100
+++ new/util/pl/VC-32.pl	2009-03-11 00:14:08.175454900 +0100
@@ -16,10 +16,10 @@
 	$crypto="libeay32";
 	}
 
-$o='\\';
+$o='/';
 $cp='$(PERL) util/copy.pl';
 $mkdir='$(PERL) util/mkdir-p.pl';
-$rm='del /Q';
+$rm='rm -f';
 
 $zlib_lib="zlib1.lib";
 
@@ -150,6 +150,7 @@
 # generate symbols.pdb unconditionally
 $app_cflag.=" /Zi /Fd\$(TMP_D)/app";
 $lib_cflag.=" /Zi /Fd\$(TMP_D)/lib";
+if ($debug) { $lib_cflag.="d"; }
 $lflags.=" /debug";
 
 $obj='.obj';
