=== modified file 'dwarf/dwarf_arange.c'
--- dwarf/dwarf_arange.c	2011-10-10 20:34:10 +0000
+++ dwarf/dwarf_arange.c	2011-10-10 20:35:13 +0000
@@ -117,6 +117,11 @@
             arange_ptr, local_length_size,
             local_extension_size);
         arange_ptr_past_end = arange_ptr + length;
+#ifdef WIN32
+		if (length == 0) {
+			continue;
+		}
+#endif
 
 
         READ_UNALIGNED(dbg, version, Dwarf_Half,

